scripts/sync_data_with_remote_server.py

```diff
@@ -0,0 +1,158 @@
+            raise FileNotFoundError
+
+        # Get the archived filename
+        upload_filename = self._get_archive_filename(filename)
+
+        # # Make sure the upload directory exists
+        # self._check_destination_directory()
+        # os.makedirs(os.path.dirname(upload_filename), exist_ok=True)
+
+        # Move the file to the DC directory
+        self.logger.debug(f"Moving {filename} to {upload_filename}.")
+        self.transfer_data(filename, upload_filename)
+
+        # Finally, delete the original
+        os.remove(filename)
+
+    def _setup_sftp(self):
+        self.ssh = pm.SSHClient()
+        self.ssh.set_missing_host_key_policy(pm.AutoAddPolicy())
+        self.ssh.connect(
+            self.remote_host,
+            username=self.username,
+            password=self.password,
+            port=self.port,
+        )
+        self.sftp = self.ssh.open_sftp()
+
+    def transfer_data(self, filename, destination):
+        self._setup_sftp()
+        try:
+            self.logger.info(
+                "Checking whether the provided destination directory exists"
+            )
+            self.sftp.stat(os.path.dirname(destination))
+            self.logger.info("The destination directory exisets in the cloud!")
+        except FileNotFoundError:
+            self.logger.info("The destination directory was not found!")
+            self.logger.info(f"Creating {os.path.dirname(destination)} in the cloud.")
+            folders = os.path.dirname(destination).split("/")
+            for folder in folders:
+                try:
+                    self.sftp.chdir(folder)
+                except FileNotFoundError:
+                    # Create the folder if it does not exist
+                    self.sftp.mkdir(folder)
+                    self.sftp.chdir(folder)
+            # self.sftp.mkdir(os.path.dirname(destination))
+        self.logger.info(
+            f"Copying {filename} into the destination directory: {destination}"
+        )
+        self.sftp.put(filename, destination)
+        self.logger.info("Copying completed!")
+        self.sftp.close()
+        self.ssh.close()
+
+if __name__ == "__main__":
+    remote_host = "remote_host"
+    username = "username"
+    password = "password"
+    port = int("port")
+    images_directory = "images_directory"
+    archive_directory = "path/to/huntsman/on/dc-cloud"
+
+    delay_interval = 2 * u.second
+    sleep_interval = 3 * u.second
+    archiver = RemoteArchiver(
+        delay_interval=delay_interval,
+        sleep_interval=sleep_interval,
+        images_directory=images_directory,
+        archive_directory=archive_directory,
+        remote_host=remote_host,
+        username=username,
+        password=password,
+        port=port,
+    )
+    archiver.start()
+
+    # Monitor the archiver
+    # If it breaks we want to terminate the script so docker can restart the service
+    while True:
```

Comment:
Can you add a `try` that gracefully handles a Keyboardinterrupt. What if this happens during a file transfer?

Reply:
Oh! I forgot to add the `try` part! will do that in the next commit when receive your comments.